### PR TITLE
Add Loading Indicators

### PR DESCRIPTION
**Pull request by conradbeach**
# Goal
Pivotal Tracker: [#172251053](https://www.pivotaltracker.com/story/show/172251053), [#172251090](https://www.pivotaltracker.com/story/show/172251090)

This branch adds loading indicators that are displayed while a search request is executing or while an update request is executing.

We also plan on adding a flash message indicating when a user is updated successfully, but that will happen in a separate pull request.

## Tradeoffs & Alternatives
The loader I used is pretty basic. I used it because it was already in the project and it suits our purposes well enough. If we want to use a fancier Atomic Jolt loader; we could switch to that, but I wasn't sure it was worth the effort

## Demo
loaders

### CODE REVIEW

File: client/apps/user_tool/reducers/application.spec.js [hunk 1]
```diff
@@ -2,7 +2,12 @@ import applicationReducer from './application';
 import { Constants as ApplicationConstants } from '../actions/application';
 
 describe('application reducer', () => {
-  const initialState = () => ({ matchingUsers: [], currentPage: 1 });
+  const initialState = (opts) => ({
```
[the remark this answers is not in the document]
Comment (conradbeach) [reply client/apps/user_tool/reducers/application.spec.js]: Done

**Comment by ThinkBrandonThink** [comment]
That's the type of loader I would have recommended. AU doesn't want to see our branding anywhere, so our other loader wouldn't have been a good fit.

**Comment by conradbeach** [comment]
Gotcha. Thanks.

**Comment by conradbeach** [comment]
Thanks for the review.

**Pull request merged**